File: packages/python/src/endpoints/repository_providers.py
"""
Repository_Providers API Client
"""

from typing import Dict, List, Optional, Any
import httpx


class Repository_ProvidersClient:
    """Client for repository_providers endpoints."""

    def __init__(self, base_url: str, client: Optional[httpx.AsyncClient] = None):
        self.base_url = base_url.rstrip('/')
        self.client = client

    async def _request(self, method: str, path: str, **kwargs) -> Any:
        """Make HTTP request."""
        url = self.base_url + path
        if self.client:
            response = await self.client.request(method, url, **kwargs)
            response.raise_for_status()
            return response.json()
        else:
            async with httpx.AsyncClient() as client:
                response = await client.request(method, url, **kwargs)
                response.raise_for_status()
                return response.json()


    async def get_root(self, enabled_only: Optional[bool] = None):
        """GET /"""
        url = f"/"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def get_provider_id(self, provider_id: str):
        """GET /{provider_id}"""
        url = f"/{provider_id}"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)

File: packages/typescript/src/endpoints/repository_providers.ts
/**
 * Repository_Providers API Client
 */

import { AxiosInstance } from 'axios'

export class Repository_ProvidersClient {
  private httpClient: AxiosInstance

  constructor(private baseURL: string, httpClient: AxiosInstance) {
    this.httpClient = httpClient
  }

  /**
   * GET /
   */
  async getRoot(enabled_only?: boolean): Promise<List[RepositoryProviderResponse]> {
        const params = {'enabled_only: enabled_only'}
        const data = undefined
        return this.httpClient.get(`/`, params)
  }
  /**
   * GET /{provider_id}
   */
  async get{provider_id}(provider_id: string): Promise<RepositoryProviderResponse> {
        const params = {'provider_id: provider_id'}
        const data = undefined
        return this.httpClient.get(`/{provider_id}`, params)
  }
}
